movement/movement.c

```diff
@@ -691,3 +713,81 @@ void cb_tick(void) {
         movement_state.subsecond++;
     }
 }
+
+/** Figures out the next scheduled event for the HPT to wake up for.
+ * Must be called whenever:
+ * - a new event is scheduled
+ * - an old event is deleted
+ * - an event is triggered
+ * - the timer overflows
+ */
+static void _movement_hpt_schedule_next_event() {
+    uint64_t next = UINT64_MAX;
+    for(uint8_t req_idx = 0; req_idx < MOVEMENT_NUM_FACES; ++req_idx) {
+        uint64_t event = hpt_scheduled_events[req_idx];
+        if(event < next) {
+            event = next;
+        }
+    }
```

Comment:
It's critical that we make this function as efficient as possible. Hopefully we can eliminate this `O(n)` loop. What's your take on [radix timers]?

[radix timers]: https://github.com/romanf-dev/timers

Reply:
I don't think it's fair to call this O(n), because MOVEMENT_NUM_FACES is a constant. Most people are probably going to have like, 12 faces max on their watch, and even if they go hog wild, the most they can have is 256. Compilers are very good at optimizing simple loops like this, it will be fine.

radix timers seems like a really "clever" solution to this very simple problem.